sphinx/transforms/__init__.py

```diff
@@ -327,3 +329,33 @@ def visit_pending_xref(self, node):
     def visit_image(self, node):
         # type: (nodes.Node) -> None
         raise nodes.SkipNode
+
+
+class SphinxSmartQuotes(DocutilsSmartQuotes):  # NOQA
+    """
+    Customized SmartQuotes to avoid transform for some extra node types.
+    """
+    def get_tokens(self, txtnodes):
+        # A generator that yields ``(texttype, nodetext)`` tuples for a list
+        # of "Text" nodes (interface to ``smartquotes.educate_tokens()``).
+
+        texttype = {True: 'literal',  # "literal" text is not changed:
+                    False: 'plain'}
+        for txtnode in txtnodes:
+            nodetype = texttype[isinstance(txtnode.parent,
+                                           (nodes.literal,
+                                            nodes.literal_block,
+                                            addnodes.literal_emphasis,
+                                            addnodes.literal_strong,
+                                            addnodes.desc_signature,
+                                            addnodes.productionlist,
+                                            addnodes.desc_optional,
+                                            addnodes.desc_name,
```

Comment:
complement: the ``Cmdoption`` inherits from ``ObjectDescription``. The latter adds ``desc`` node, and then a ``desc_signature``. The ``Cmdoption`` adds in particular ``desc_name`` nodes. The ``pseudoxml`` produces something like this (I removed the ``.. program::`` directive from the minimal example):

```
        <desc desctype="option" domain="std" noindex="False" objtype="option">
            <desc_signature allnames="-h --help --version" first="False" ids="cmdoption-h" names="cmdoption-h cmdoption--help cmdoption--version">
                <desc_name>
                    -h
                <desc_addname>
                <desc_addname>
                    , 
                <desc_name>
                    --help
                <desc_addname>
                <desc_addname>
                    , 
                <desc_name>
                    --version
                <desc_addname>
            <desc_content>
                <paragraph>
                    Display usage summary or Sphinx version.
```

The ``visit_option()`` and ``depart_option()`` in  Sphinx 1.5.6 ``SmartyPantsHTMLTranslator`` appear to have never been actually executed.